apps/services/license-api/src/app/modules/license/guards/licenseTypeScope.guard.ts

```diff
@@ -1,40 +1,43 @@
 import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common'
 import { Observable } from 'rxjs'
 import { Reflector } from '@nestjs/core'
-import { SCOPES_KEY, getRequest } from '@island.is/auth-nest-tools'
+import { getRequest } from '@island.is/auth-nest-tools'
+import { LicenseApiScope, getLicenseTypeScopes } from '@island.is/auth/scopes'
+import { licenseTypeToScope } from '../scopeMapper'
+import { LicenseId } from '../license.types'
 
 @Injectable()
 export class LicenseTypeScopesGuard implements CanActivate {
-  constructor(private readonly reflector: Reflector) {}
-
   canActivate(
     context: ExecutionContext,
   ): boolean | Promise<boolean> | Observable<boolean> {
-    const scopes = this.reflector.getAllAndOverride<string[]>(SCOPES_KEY, [
-      context.getHandler(),
-      context.getClass(),
-    ])
-
     const request = getRequest(context)
 
-    if (scopes && !this.hasScope(scopes, request.auth?.scope)) {
+    const licenseId = request.params.licenseId
+    const isVerify = licenseId === 'verify'
+
+    if (!licenseId) {
       return false
     }
 
-    const licenseType = this.getLicenseTypeFromUrl(request.url)
-    const scopeActions = this.getScopeActions(request.auth?.scope ?? [])
+    const scopes = isVerify
+      ? [LicenseApiScope.licensesVerify]
+      : getLicenseTypeScopes()
+
+    const authScope = request.auth?.scope
+
+    if (scopes && !this.hasScope(scopes, authScope)) {
+      return false
+    }
```

Comment:
This extra check (and `getLicenseTypeScopes()` is strictly not neccessary as the check below should cover everything.